icu4j/main/classes/core/src/com/ibm/icu/impl/number/DecimalQuantity.java

```diff
@@ -219,4 +226,15 @@
      *            The {@link UFieldPosition} to populate.
      */
     public void populateUFieldPosition(FieldPosition fp);
+
+    /**
+     * @return Whether this {@link DecimalQuantity} has been put into compact notation.
+     */
+    public boolean hasCompactNotation();
+
+    /**
+     * Sets whether this instance has been put into compact notation.
+     * @param hasCompactNotation New value for whether
+     */
+    public void setHasCompactNotation(boolean hasCompactNotation);
```

Comment:
I think a boolean option is not the right approach since you need to remember whether the magnitude was changed as a result of `scale()` or `notation()`.  For example, you could have a call site like,

```java
NumberFormatter.with().scale(Scale.powerOfTen(2)).notation(Notation.compactShort())
```

Consider adding a new field `exponent` (or `suppressedExponent`, etc) and add an argument to `adjustMagnitude` that controls whether to record the change in `exponent` or not.

Reply:
Done.  Using `exponent` field, and `adjustMagnitude` was updated accordingly.